src/dc_e2ee.rs

```diff
@@ -183,19 +185,19 @@ pub unsafe fn dc_e2ee_encrypt(
                             if iCnt > 1i32 {
                                 let mut i: libc::c_int = 0i32;
                                 while i < iCnt {
-                                    let p: *mut libc::c_char = dc_apeerstate_render_gossip_header(
-                                        &peerstates[i as usize],
-                                        min_verified,
-                                    );
-                                    if !p.is_null() {
+                                    let p = peerstates[i as usize]
+                                        .render_gossip_header(min_verified as usize);
+
+                                    if p.is_some() {
+                                        let header = to_cstring(p.unwrap());
                                         mailimf_fields_add(
                                             imffields_encrypted,
                                             mailimf_field_new_custom(
                                                 strdup(
                                                     b"Autocrypt-Gossip\x00" as *const u8
                                                         as *const libc::c_char,
                                                 ),
-                                                p,
+                                                strdup(header.as_ptr()),
                                             ),
```

Comment:
Does this leak or does `mailimf_field_new_custom()` clean it up?

Reply:
I believe it gets freed when the structure wrapping it that mailimf_field_new_custom creates gets freed